lib/real_world/feed/listeners/tracker.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'colmena/listener'
+
+module RealWorld
+  module Feed
+    module Listeners
+      class Tracker < Colmena::Listener
+        def call(event)
+          case event.fetch(:type)
+          when :article_created
+            followers = port(:router).query(:list_followers).call(
+              followed_id: event.dig(:data, :author_id),
+            ).fetch(:data)
```

Comment:
How do we deal with pagination here?

Reply:
I think I'd simply iterate the pages. We can introduce a more sophisticated mechanism in the future and write an article about it too.

Here are my thoughts though:

* This is an asynchronous listener, so we're not blocking anything.
* We don't need to respect strictly the order in which articles are pushed to a feed (if we push first an article that was published 2 minutes later, the UX won't suffer).
* Thus, we can scale this listener horizontally as much as we want (processing articles 100% concurrently)
* This means, we wouldn't care if it took 2 whole minutes to push an article to all follower feeds.
* That said, if we had potentially long async tasks, it would be good to have a mechanism to save the current state every few seconds and recover from the latest checkpoint if the task fails.

I think that would be an interesting discussion with the community. I'm also curious about how would the feed work in something like Twitter, where it seems to be calculated and parametrized according to tweet popularity and so on.